### DISCUSSION TITLE
Can't change delay between last and first frames

### DISCUSSION THREAD

**Discussion by camerooncameroon** [ImageMagick/ImageMagick#5649] [Help] [answered]
Hi, I need to convert 5000+ gif files so that each of them pauses after each sequence and before starting a new one (loops shall be endless). Number of frames in each gif file is different.
I use this code:
```
@ECHO OFF
CHCP 65001
PUSHD %~dp0

SET magick="c:\Program Files\ImageMagick-7.1.0-Q16-HDRI\magick.exe"

FOR %%f IN (*.gif) DO (
  %magick% convert "%%f" -delay 100 -loop 0 "%%~nf_new.gif"
  )
```
No errors are reported after executing this bat file and new gif files are created.
But the delay stays the same (it was about 40 seconds, and I need to change it to 1-2 secs no more). 

What am I doing wrong, how the command line shall be written correctly?

An example file is attached.

![6ca6](https://user-images.githubusercontent.com/78047188/195066016-cbc736b2-9dbc-45d1-bcf3-4db40278337f.gif)

**Comment by snibgo** [comment, marked as answer]
You want to set the delay on only one frame, not all frames.

"-delay" is used to set the delay on all following frames. "-set delay" is used to set the delay in all frames already in the current list.

```
magick -loop 0 -delay 0 ideagr.gif -loop 0 ( +clone -set delay 500 ) -delete -2 out.gif
```
... where ideagr.gif is your input gif.

**Comment by camerooncameroon** [comment]
@snibgo 
Thanks! This works perfectly. Even though I don't understand the whole magick it does )

**Comment by snibgo** [comment]
The second "-loop 0" is not required, sorry, so the command should be:
```
magick -loop 0 -delay 0 ideagr.gif ( +clone -set delay 500 ) -delete -2 out.gif
```
"-loop 0" means that following frames will loop forever.

"-delay 0" means that following frames will have no delay after each is shown, before the next is shown.

"ideagr.gif" reads your input GIF, so each frame is an image in the list..

"(" starts a new empty list.

"+clone" copies the last frame from the outer list into this (empty) list.

"-set delay 500" sets the delay after this frame to about 5 seconds.

")" closes the current list, copying the frame with the long delay to the end of the outer list.

Now the last frame has two copies, and we only want the last one, so "-delete -2" removes the second from last image.

"out.gif" writes all the frames to a GIF file.

**Comment by camerooncameroon** [comment]
Thanks again! A bit (a lot) more complicated than I expected initially. But now, thanks to you, all is working. BTW, even the first line you suggested (with the excess -loop option), also provided the desired results.
